libvirt/resource_libvirt_domain.go

```diff
@@ -602,6 +602,18 @@ func resourceLibvirtDomainCreate(ctx context.Context, d *schema.ResourceData, me
 	d.SetId(id)
 	log.Printf("[INFO] Domain ID: %s", d.Id())
 
+	if d.Get("qemu_agent").(bool) && d.Get("running").(bool) {
```

Comment:
Why does the code check for `d.Get("qemu_agent")` if then `waitingForAgentRunning` checks again?